app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,51 @@
+require_relative "../lib/oidc_client.rb"
+
+class SessionsController < ApplicationController
+  def create
+    if logged_in?
+      redirect_to redirect_path
+      return
+    end
+
+    if Rails.env.test?
+      render plain: "Redirecting to login"
+    else
+      redirect_to oidc.auth_uri[:uri]
+    end
```

Comment:
We just do the redirect and check that happens in the tests

Reply:
I was having some trouble getting this to run in the tests, possibly around the redirect to an external URL.  I'll see if I can reproduce this to come up with a more elegant solution.

Reply:
I remember now.  Capybara ignores the root of the redirect, so a request to (say) `http://www.login.service.dev.gov.uk/login` will actually get redirected to `http://finder-frontend.dev.gov.uk/login`, which then results in a query being made to `http://content-store.dev.gov.uk/content/login` (because of the "catch-all" wildcard in routes).  I found a little bit of explanation in https://makandracards.com/makandra/15217-test-redirects-to-an-external-url-with-cucumber-capybara, but could't come up with a more elegant solution.